ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ranger/RangerRestClientImpl.java

```diff
@@ -349,6 +349,70 @@ public boolean checkConnection(String url) {
     return (clientResp.getStatus() < HttpServletResponse.SC_UNAUTHORIZED);
   }
 
+  @Override
+  public List<RangerPolicy> addDenyPolicies(List<RangerPolicy> rangerPolicies, String rangerServiceName,
+                                            String sourceDb, String targetDb) throws SemanticException {
+    if (StringUtils.isEmpty(rangerServiceName)) {
+      throw new SemanticException("Ranger Service Name cannot be empty");
+    }
+    RangerPolicy denyRangerPolicy = new RangerPolicy();
+    denyRangerPolicy.setService(rangerServiceName);
+    denyRangerPolicy.setName(sourceDb + "_replication deny policy for " + targetDb);
+    Map<String, RangerPolicy.RangerPolicyResource> rangerPolicyResourceMap = new HashMap<String,
+        RangerPolicy.RangerPolicyResource>();
+    RangerPolicy.RangerPolicyResource rangerPolicyResource = new RangerPolicy.RangerPolicyResource();
+    List<String> resourceNameList = new ArrayList<String>();
+
+    List<RangerPolicy.RangerPolicyItem> denyPolicyItemsForPublicGroup = denyRangerPolicy.getDenyPolicyItems();
+    RangerPolicy.RangerPolicyItem denyPolicyItem = new RangerPolicy.RangerPolicyItem();
+    List<RangerPolicy.RangerPolicyItemAccess> denyPolicyItemAccesses = new ArrayList<RangerPolicy.
+        RangerPolicyItemAccess>();
+
+    List<RangerPolicy.RangerPolicyItem> denyExceptionsItemsForBeaconUser = denyRangerPolicy.getDenyExceptions();
+    RangerPolicy.RangerPolicyItem denyExceptionsPolicyItem = new RangerPolicy.RangerPolicyItem();
+    List<RangerPolicy.RangerPolicyItemAccess> denyExceptionsPolicyItemAccesses = new ArrayList<RangerPolicy.
+        RangerPolicyItemAccess>();
+
+    resourceNameList.add(sourceDb);
+    rangerPolicyResource.setValues(resourceNameList);
+    RangerPolicy.RangerPolicyResource rangerPolicyResourceColumn =new RangerPolicy.RangerPolicyResource();
+    rangerPolicyResourceColumn.setValues(new ArrayList<String>(){{add("*"); }});
+    RangerPolicy.RangerPolicyResource rangerPolicyResourceTable =new RangerPolicy.RangerPolicyResource();
+    rangerPolicyResourceTable.setValues(new ArrayList<String>(){{add("*"); }});
+    rangerPolicyResourceMap.put("database", rangerPolicyResource);
+    rangerPolicyResourceMap.put("table", rangerPolicyResourceTable);
+    rangerPolicyResourceMap.put("column", rangerPolicyResourceColumn);
+    denyRangerPolicy.setResources(rangerPolicyResourceMap);
+
+    List<String> accessTypes = new ArrayList<>();
+    accessTypes.addAll(Arrays.asList("create", "update", "drop", "alter", "index", "lock", "write", "ReplAdmin"));
```

Comment:
Don't need to do addAll. You can directly use List<String> accessTypes = Arrays.asList